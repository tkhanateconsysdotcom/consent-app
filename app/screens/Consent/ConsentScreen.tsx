import React from "react";
import { ScrollView } from "react-native";
import { useSelector } from "react-redux";
import {
  AdMobBanner,
  AdMobInterstitial,
  PublisherBanner,
  AdMobRewarded,
} from "react-native-admob";

import { Text, Button } from "components";
import { t } from "utils";
import { navigate } from "navigation";

import styles from "./ConsentScreen.styles";
import { NumberOfDaysVegetarian, ProgressChart } from "./components";
import { selectors } from "./ducks";
import navigationOptions from "./ConsentScreen.navigationOptions";

interface Props {
  navigation: {
    push: (screen: string) => void;
  };
}

const ConsentScreen = (props: Props) => {
  const navigator = navigate(props.navigation);
  const monthlyAcceptanceConsent = useSelector(
    selectors.getMonthlyAcceptanceConsent
  );
  const totalCurrentMonthPartners = useSelector(
    selectors.getCurrentMonthAllAcceptanceValue
  );
  const transportCurrentMonthPartners = useSelector(
    selectors.getCurrentMonthTransportAcceptanceValue
  );
  const foodCurrentMonthPartners = useSelector(
    selectors.getCurrentMonthFoodAcceptanceValue
  );
  const otherCurrentMonthPartners = useSelector(
    selectors.getCurrentMonthOtherAcceptanceValue
  );

  const totalCurrentYearPartners = useSelector(
    selectors.getCurrentYearAllAcceptanceValue
  );
  const transportCurrentYearPartners = useSelector(
    selectors.getCurrentMonthTransportAcceptanceValue
  );
  const foodCurrentYearPartners = useSelector(
    selectors.getCurrentYearFoodAcceptanceValue
  );
  const otherCurrentYearPartners = useSelector(
    selectors.getCurrentYearOtherAcceptanceValue
  );

  // Display an interstitial
  AdMobInterstitial.setAdUnitID("ca-app-pub-4482303808779676~5440604106");
  AdMobInterstitial.setTestDevices([AdMobInterstitial.simulatorId]);
  AdMobInterstitial.requestAd().then(() => AdMobInterstitial.showAd());

  // Display a rewarded ad
  AdMobRewarded.setAdUnitID("ca-app-pub-4482303808779676~5440604106");
  AdMobRewarded.requestAd().then(() => AdMobRewarded.showAd());

  return (
    <ScrollView style={styles.container}>
      <AdMobBanner
        adSize="fullBanner"
        adUnitID="ca-app-pub-4482303808779676~5440604106"
        testDevices={[AdMobBanner.simulatorId]}
        onAdFailedToLoad={(error) => console.error(error)}
        style={{ width: "auto", height: "auto" }}
      />
      <PublisherBanner
        adSize="fullBanner"
        adUnitID="ca-app-pub-4482303808779676~5440604106"
        testDevices={[PublisherBanner.simulatorId]}
        onAdFailedToLoad={(error) => console.error(error)}
        onAppEvent={(event) => console.log(event.name, event.info)}
        style={{ width: "auto", height: "auto" }}
      />
      <ProgressChart
        isMonth
        totalPartners={totalCurrentMonthPartners}
        foodPartners={foodCurrentMonthPartners}
        transportPartners={transportCurrentMonthPartners}
        otherPartners={otherCurrentMonthPartners}
        monthlyPartnersConsent={monthlyAcceptanceConsent}
      />
      <Button.Primary
        style={styles.monthlyConsentButton}
        fullWidth
        textType={"Primary"}
        onPress={() => navigator.openMontlyConsent()}
      >
        <Text.Primary bold center white>
          {t("CONSENT_SCREEN_SET_MONTHLY_CONSENT")}
        </Text.Primary>
      </Button.Primary>
      <NumberOfDaysVegetarian />
      <ProgressChart
        totalPartners={totalCurrentYearPartners}
        foodPartners={foodCurrentYearPartners}
        transportPartners={transportCurrentYearPartners}
        otherPartners={otherCurrentYearPartners}
        monthlyPartnersConsent={monthlyAcceptanceConsent}
      />
    </ScrollView>
  );
};

ConsentScreen.navigationOptions = navigationOptions;

export default ConsentScreen;
