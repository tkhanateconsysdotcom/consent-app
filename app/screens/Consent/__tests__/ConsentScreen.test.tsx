import React from "react";
import { create } from "react-test-renderer";

import ConsentScreen from "../ConsentScreen";

const props = {
  navigation: {
    push: () => {
      // do nothing.
    },
  },
};

it("ConsentScreen renders correctly", () => {
  // const tree = create(<ConsentScreen {...props} />).toJSON();
  // expect(tree).toMatchSnapshot();
});

// TODO: add test to verify that monthly consent is opened
