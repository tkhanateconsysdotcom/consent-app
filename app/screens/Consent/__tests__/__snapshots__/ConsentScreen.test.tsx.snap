// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsentScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <ProgressChart
      foodPartners={[MockFunction]}
      isMonth={true}
      monthlyPartnersConsent={[MockFunction]}
      otherPartners={[MockFunction]}
      totalPartners={[MockFunction]}
      transportPartners={[MockFunction]}
    />
    <Button.Primary
      fullWidth={true}
      onPress={[Function]}
      style={
        Object {
          "marginVertical": 6,
        }
      }
      textType="Primary"
    >
      <Text.Primary
        bold={true}
        center={true}
        white={true}
      >
        Set monthly consent
      </Text.Primary>
    </Button.Primary>
    <NumberOfDaysVegetarian />
    <ProgressChart
      foodPartners={[MockFunction]}
      monthlyPartnersConsent={[MockFunction]}
      otherPartners={[MockFunction]}
      totalPartners={[MockFunction]}
      transportPartners={[MockFunction]}
    />
  </View>
</RCTScrollView>
`;
